app/scripts/modules/core/src/presentation/forms/fields/FormikFormField.tsx

```diff
@@ -70,13 +82,56 @@ function FormikFormFieldImpl<T = any>(props: IFormikFormFieldImplProps<T>) {
     messageNode,
   };
 
+  const [inputMode, setInputMode] = React.useState(SpelAwareInputMode.DEFAULT);
+
+  const modeToInput: { [k in SpelAwareInputMode]: React.ComponentType<IFormInputProps> } = {
+    [SpelAwareInputMode.DEFAULT]: input,
+    [SpelAwareInputMode.FREEFORM]: SimpleSpelInput,
+  };
+
+  const freeformInputEnabled = firstDefined(
+    get(props, 'spelConfig.freeformInputEnabled', null),
```

Comment:
To get typings to flow through, we can use optional chaining.  Inspired by @anotherchrisberry  in #7742
```suggestion
    props.spelConfig?.freeformInputEnabled ?? null;
```

Reply:
Amaaaaazing thank you for the inspo @anotherchrisberry